x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/rules_table_toolbar.tsx

```diff
@@ -26,6 +26,22 @@ export const RulesTableToolbar = React.memo(() => {
     (ruleManagementFilters?.rules_summary.prebuilt_installed_count ?? 0);
   const updateTotal = prebuiltRulesStatus?.num_prebuilt_rules_to_upgrade ?? 0;
 
+  const ruleUpdateTab = useMemo(
```

Comment:
Nit: extracting it into a separate memoized object is probably excessive, because `useMemo` that returns `ruleTabs` still depends on `updateTotal`.

```ts
  const ruleTabs = useMemo(
    () => ({
      [AllRulesTabs.management]: {
        id: AllRulesTabs.management,
        name: i18n.INSTALLED_RULES_TAB,
        disabled: false,
        href: `/rules/${AllRulesTabs.management}`,
        isBeta: installedTotal > 0,
        betaOptions: {
          text: `${installedTotal}`,
        },
      },
      [AllRulesTabs.monitoring]: {
        id: AllRulesTabs.monitoring,
        name: i18n.RULE_MONITORING_TAB,
        disabled: false,
        href: `/rules/${AllRulesTabs.monitoring}`,
        isBeta: installedTotal > 0,
        betaOptions: {
          text: `${installedTotal}`,
        },
      },
      ...(updateTotal > 0
        ? {
            [AllRulesTabs.updates]: {
              id: AllRulesTabs.updates,
              name: i18n.RULE_UPDATES_TAB,
              disabled: false,
              href: `/rules/${AllRulesTabs.updates}`,
              isBeta: updateTotal > 0,
              betaOptions: {
                text: `${updateTotal}`,
              },
            },
          }
        : {}),
    }),
    [installedTotal, updateTotal]
  );
```